### ISSUE TITLE
Accordion feature fails accessibility tests

### ISSUE THREAD

**Issue by kstack7** [girldevelopit/gdi-website#339]
**Description**
_The accordion list of chapters (small screen/mobile view of /chapters) gets buggy for screen readers. The CSS ends up hiding or overlapping the chapters instead of tabbing through them in a linear fashion. See screenshot for example._

**Ask**
_Implement a new list design that is easy-to-read on mobile and accessible for screen readers._

**To Do**
- [x] Remove accordion-based code on view/chapters/index.html.erb
- [x] Add @Roenok's recommended progressive collapsibles code
- [x] Update chapters.css.scss to reflect new naming conventions
- [x] Test screen reader / tabbing 
- [x] Enjoy new happy state of accessibility!

**Additional Info**
Link to recommended code: http://heydonworks.com/practical_aria_examples/#progressive-collapsibles
Example of wonky existing code:
![screen shot 2015-07-23 at 4 44 53 pm](https://cloud.githubusercontent.com/assets/6612770/8861956/4db5e152-315b-11e5-8381-1cfee273f8d3.png)

**Comment by kstack7** [comment]
I'm putting this up while working on it btw!

**Comment by kstack7** [comment]
I currently have the code swapped out and looks like this when tabbing through:

![screen shot 2015-07-23 at 4 56 15 pm](https://cloud.githubusercontent.com/assets/6612770/8862041/ca8d15e2-315b-11e5-91d4-056174c5a8dd.png)

**Comment by kstack7** [comment]
@Roenok what types of things should I look for with this new code in place?

**Comment by Roenok** [comment]
@kstack Set your mouse to the side/disable the touchpad on your laptop. Can you still navigate the list using tab (move forward) shift+tab (move back) and enter and/or spacebar (trigger)? If you can do that, there is a good chance it will work with assistive technology too.

**Comment by kstack7** [comment]
@Roenok  Yes'm! 'Cept I notice space bar doesn't trigger the same way as 'enter' does. Like I can only go to the page I want if I hit enter on the links; space bar sends me to the bottom of the page. Is this right?

**Comment by kstack7** [comment]
I can pop this on staging if it helps to see...

**Comment by Roenok** [comment]
@kstack7 That's expected behavior, yes. Space is used as a trigger on places where enter would cause problems, like to select a checkbox on forms (because enter might accidentally send)

Pop it on staging and I'll run a full test with a couple of screen readers.

**Comment by kstack7** [comment]
Okay -- here it is on staging...hack away

https://girl-develop-it-staging.herokuapp.com/chapters

**Comment by Roenok** [comment]
@kstack7 On staging I am seeing two nested buttons for each h3, which causes weird usability issues. Is the JS malfunctioning?
![double-button](https://cloud.githubusercontent.com/assets/925963/8894489/54b2d28a-3386-11e5-864c-e0ff59cb026a.PNG)

**Comment by kstack7** [comment]
Hah! Good point, @Roenok ...I was noticing some duplications in the subsequent div but not the buttons...let me see what's up...thanks!

**Comment by kstack7** [comment]
Sooo main.js is getting loaded twice! fun fact, huh!...I'll see which library is adding it in aside from the rails layout...we never noticed this before because the old accordion code did not wrap/alter the HTML at all like this new stuff does. 'Least we can see why!

**Comment by kstack7** [comment]
Ugh. Y'know. I think this is due to weird wall I hit while deploying staging from a cafe the other day. I noticed it seemed hung up and tried to redeploy all while on this weak connection.

I just redeployed once more and everything looks happy now haha sooooo try it now?

**Comment by Roenok** [comment]
@kstack7 Still seeing double, alas

**Comment by kstack7** [comment]
Gah! It was definitely gone after the last deploy..okay, I may need some extra set of eyes on this since it's being touchy.

**Comment by kstack7** [comment]
Okay taking another look. I think this has something to do with the different environment settings/behaviors since this is not happening on production. It's the precompiled js files mixing with the ones loaded into the layout. Anyone seeing the obvious here that I'm missing?

**Comment by kstack7** [comment]
Mmm I take that back because the rails environment is still 'production' for staging...hmmmm...

**Comment by kstack7** [comment]
K so...taking main.js out of the application.html.erb eliminated the duplications since it is successfully compiled/minified in the application.js file...

Can anyone shed light as to why we want to call these compiled assets into the layout? Seems like this has been going on, but we just never noticed the double load since we weren't manipulating the DOM like these dropdown do.

Commented out main.js (it only has the dropdown code right now anyways):

![screen shot 2015-07-29 at 1 19 35 am](https://cloud.githubusercontent.com/assets/6612770/8950483/26bbf1de-3590-11e5-9140-938b822fd9d4.png)

And now no duplication since application.js has all the JS files in it and thus the main.js/dropdown code is rendered only once: 

![screen shot 2015-07-29 at 1 20 08 am](https://cloud.githubusercontent.com/assets/6612770/8950490/52a48400-3590-11e5-81d8-c6cfe40768f8.png)

Shouldn't we just stick to the minified application.js file?

**Comment by Roenok** [comment]
@kstack7 The minified file sounds like a good idea to me. Deploy to staging so I can test?

**Comment by kstack7** [comment]
Woo! Way happier:

https://girl-develop-it-staging.herokuapp.com/chapters

**Comment by Roenok** [comment]
Works great with keyboard-only. Thanks!

**Comment by kstack7** [comment]
Fixed with #353
